### PR TITLE
[improvement](stream-load) add http request field check for stream load in BE

### PR DESCRIPTION
**Pull request by jixxiong**
## Proposed changes

Issue Number: close #xxx

Currently, when users use stream load, any misspelled fields in the request will be ignored, and the import process will continue. This may lead to inconsistent import results with the users' expectations.

Therefore, the purpose of this code is to halt the import process immediately and throw an error if there are any misspelled fields during the stream load. It will not proceed with the import process if there are erroneous fields.

This functionality is controlled by the backend option `check_stream_load_request_fields`, which is set to false by default. If it is set to true, during the stream load import, the code will validate each field in the header to check its validity. If any field is found to be invalid, the import process will be halted, and an error will be reported.

For example, when `check_stream_load_request_fields` is set to true, if we use an invalid request field such as `i_am_not_a_valid_field` in stream load, the result would be:

```
{
    ...
    "Status": "Fail",
    "Message": "[INVALID_ARGUMENT]http request field `i_am_not_a_valid_field` not support in stream load.",
    ...
}
```

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/http/action/stream_load.cpp [hunk 1]
```diff
@@ -137,6 +137,120 @@ static bool is_format_support_streaming(TFileFormatType::type format) {
     }
 }
 
+const std::unordered_set<std::string> StreamLoadAction::kStreamLoadAvailableRequestFields {
```
Comment (zhannngchen) [inline be/src/http/action/stream_load.cpp:140]: move to http_common.h, should be static

**Comment by zhannngchen** [comment]
Add some regression test

**Review by zhannngchen — DISMISSED** [review]
LGTM

**Review by zhannngchen — DISMISSED** [review]
LGTM

**Comment by zhannngchen** [comment]
run buildall

**Comment by zhannngchen** [comment]
run buildall

File: be/src/common/config.cpp [hunk 2]
```diff
@@ -1041,6 +1041,9 @@ DEFINE_mInt64(auto_inc_low_water_level_mark_size_ratio, "3");
 // number of threads that fetch auto-inc ranges from FE
 DEFINE_mInt64(auto_inc_fetch_thread_num, "3");
 
+// wethar to check http request field for stream load in be
```
Comment (yiguolei) [inline be/src/common/config.cpp:1044]: whether

File: be/src/http/http_common.h [hunk 3]
```diff
@@ -60,4 +62,118 @@ static const std::string HTTP_TWO_PHASE_COMMIT = "two_phase_commit";
...
+        HTTP_MERGE_TYPE,
+        HTTP_DELETE_CONDITION,
+        HTTP_MAX_FILTER_RATIO,
+        HTTP_HIDDEN_COLUMNS,
+        HTTP_TRIM_DOUBLE_QUOTES,
+        HTTP_SKIP_LINES,
+        HTTP_ENABLE_PROFILE,
+        HTTP_PARTIAL_COLUMNS,
+
+        // below are http standard request fields
```
Comment (yiguolei) [inline be/src/http/http_common.h:103]: why do we need check the http headers? If http version upgrade or some headers we do not know currently, we need fix doris's code.

Comment (zhannngchen) [reply be/src/http/http_common.h:103]: user is easy to misuse the new added partial_columns header, if it's not set correctly, the load will still success, but result would not match the user's expectation.

**Comment by jixxiong** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 44.73 seconds
 stream load tsv:          507 seconds loaded 74807831229 Bytes, about 140 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.3 seconds inserted 10000000 Rows, about 341K ops/s
 storage size: 17162634405 Bytes

**Review by yiguolei — DISMISSED** [review]
LGTM

**Comment by zhannngchen** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 44.61 seconds
 stream load tsv:          512 seconds loaded 74807831229 Bytes, about 139 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          32 seconds loaded 861443392 Bytes, about 25 MB/s
 insert into select:          30.2 seconds inserted 10000000 Rows, about 331K ops/s
 storage size: 17161714853 Bytes

**Comment by jixxiong** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 46.6 seconds
 stream load tsv:          512 seconds loaded 74807831229 Bytes, about 139 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          64 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.3 seconds inserted 10000000 Rows, about 341K ops/s
 storage size: 17163019695 Bytes

**Comment by jixxiong** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 48.31 seconds
 stream load tsv:          513 seconds loaded 74807831229 Bytes, about 139 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          30 seconds loaded 861443392 Bytes, about 27 MB/s
 insert into select:          29.3 seconds inserted 10000000 Rows, about 341K ops/s
 storage size: 17162347084 Bytes

**Comment by jixxiong** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45.46 seconds
 stream load tsv:          514 seconds loaded 74807831229 Bytes, about 138 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.4 seconds inserted 10000000 Rows, about 340K ops/s
 storage size: 17162471339 Bytes

**Comment by jixxiong** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45.37 seconds
 stream load tsv:          511 seconds loaded 74807831229 Bytes, about 139 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          32 seconds loaded 861443392 Bytes, about 25 MB/s
 insert into select:          29.2 seconds inserted 10000000 Rows, about 342K ops/s
 storage size: 17162304565 Bytes

**Pull request closed without merging**